src/main/java/org/openlmis/requisition/domain/requisition/Requisition.java

```diff
@@ -1074,6 +1164,31 @@ private void updateConsumptions(Map<VersionIdentityDto, OrderableDto> orderables
     }
   }
 
+  private void updateStockConsumptions(
+      Map<VersionIdentityDto, OrderableDto> orderables,
+      List<StockCardRangeSummaryDto> stockCardRangeSummariesToAverage,
+      List<ProcessingPeriodDto> periods) {
+
+    if (template.isColumnInTemplateAndDisplayed(ADJUSTED_CONSUMPTION)) {
```

Comment:
This is more like FYI, question, not necessary a request to change.

This looks strange, `calculateAndSetStockBasedAverageConsumption` checks for `AVERAGE_CONSUMPTION` column too. If both columns are  displayed, it runs `getNonSkippedFullSupplyRequisitionLineItems` twice. 
Inside the 1st `if`, the value is calculated by some helper method and set inside RequisitionLineItem, but the 2nd `if` uses `RequisitionLineItem.calculateAndSetStockBasedAverageConsumption` method to calculate (so RequisitionLineItem sets its fields by itself) - this gives impression that `RequisitionLineItem` is a full OOP Object with methods and simple value-object at the same time, which is bad design IMO.

I was thinking about this suggestion, but it dosn't solve all concerns:
```
if(template.isColumnInTemplateAndDisplayed(ADJUSTED_CONSUMPTION) || template.isColumnInTemplateAndDisplayed(AVERAGE_CONSUMPTION))  {
  for(RequisitionLineItem line : getNonSkippedFullSupplyRequisitionLineItems(orderables)) {
    if(template.isColumnInTemplateAndDisplayed(ADJUSTED_CONSUMPTION)) {
      line.setAdjustedConsumption(...);
    }

    if(template.isColumnInTemplateAndDisplayed(AVERAGE_CONSUMPTION)) {
      StockCardRangeSummaryDto summaryToAverage = findStockCardRangeSummary(
            stockCardRangeSummariesToAverage, line.getOrderable().getId());

        line.calculateAndSetStockBasedAverageConsumption(summaryToAverage,
            template, periods, previousRequisitions);
    }
  }
}
```